d3a_interface/validators/finite_diesel_generator_validator.py

```diff
@@ -16,34 +16,39 @@
 from d3a_interface.constants_limits import ConstSettings
 from d3a_interface.exceptions import D3ADeviceException
 from d3a_interface.validators import utils
-from d3a_interface.validators.cep_validator import validate_commercial_producer
+from d3a_interface.validators.cep_validator import CommercialProducerValidator
 
 CepSettings = ConstSettings.CommercialProducerSettings
 
 
-def validate_finite_diesel_generator(**kwargs):
-    if "max_available_power_kW" in kwargs and kwargs["max_available_power_kW"] is not None:
-        if isinstance(kwargs["max_available_power_kW"], (int, float)):
-            error_message = \
-                {"misconfiguration": ["max_available_power_kW should be in between "
-                                      f"{CepSettings.MAX_POWER_KW_LIMIT.min} & "
-                                      f"{CepSettings.MAX_POWER_KW_LIMIT.max}."]}
-            utils.validate_range_limit(
-                CepSettings.MAX_POWER_KW_LIMIT.min,
-                kwargs["max_available_power_kW"],
-                CepSettings.MAX_POWER_KW_LIMIT.max,
-                error_message)
-        elif isinstance(kwargs["max_available_power_kW"], dict):
-            error_message = \
-                {"misconfiguration": ["max_available_power_kW should be in between "
-                                      f"{CepSettings.MAX_POWER_KW_LIMIT.min} & "
-                                      f"{CepSettings.MAX_POWER_KW_LIMIT.max}."]}
-            for _, value in kwargs["max_available_power_kW"].items():
+class FiniteDieselGeneratorValidator(CommercialProducerValidator):
+    """Validator class for Finite Diesel Generators."""
+
+    @classmethod
+    def validate(cls, **kwargs):
+        """Validate the parameters of the device."""
+        if kwargs.get("max_available_power_kW") is not None:
```

Comment:
You can save up an indentation level
```suggestion
        if kwargs.get("max_available_power_kW") is None:
               super().validate(**kwargs)
        if isinstance(kwargs["max_available_power_kW"], (int, float)):
```